### ISSUE TITLE
conflict in storage composer.json between specified versions of phpdocumentor and twig

### ISSUE THREAD

**Issue by scottw-finao** [opencart/opencart#13699]
**What version of OpenCart are you reporting this for?**

v4.0.2.3

**Describe the bug**

unable to run composer update in storage directory due to conflict in composer.json between phpdocumentor and twig.

What section does it affect?

storage vendor files

**To Reproduce**
1. Fresh install
  1b (create storage)
2. open command line
3. cd to storage directory
4. run composer update

**Expected behavior**
composer checks for possible updates, rebuilds autoload files, etc.

**Screenshots / Screen recordings**
```
$ composer update
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires phpdocumentor/phpdocumentor 3.3.1 -> satisfiable by phpdocumentor/phpdocumentor[v3.3.1].
    - phpdocumentor/phpdocumentor v3.3.1 requires twig/twig ~2.0 -> found twig/twig[v2.0.0, ..., v2.16.0] but it conflicts with your root composer.json require (^3.3.7).

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.

```
**Server / Test environment (please complete the following information):**
  apche 2.4.52 on ubuntu 22.0.4.4 with php 8.2.15

**Additional context**

I was looking to install another dependency to utilize .env configuration files. Before doing a composer require, I ran a simple update and it errored out as above. I have tried various versioning of phpdocumentor with no luck.

**Comment by scottw-finao** [comment]
With more digging, i think this may be related to the import/export module installed from the marketplace. Going over there to check for possible bugs listed.

**Comment by danielkerr** [comment]
any replacement out there for phpdocs since it seems its noot updated

**Comment by scottw-finao** [comment]
I did a fresh install to test without adding any extensions and confirmed it does it out-of-the-gate with a fresh install.

**Comment by scottw-finao** [comment]
I got a suggestion from the folks over at phpdocumentor:

[phpdocumentor github issues #3673](https://github.com/phpDocumentor/phpDocumentor/issues/3673)

What I basically did was replace the phpdocumenter line in composer.json with the shim require and it seems to have solved that issue, but that pops up another one after composer completes.

The vendor.php seems to be looking for guzzlehttp/promises/src/function_include.php but after running composer update, that file no longer exists in vendor/guzzlehttp/promises/src

**Comment by scottw-finao** [comment]
Out of curiosity, is there an RFC or white paper anywhere that describes why they use the vendor.php for composer files?  Now that I have that folder loading properly, I found I was able to basically comment out the entire vendor.php file and just use a single `require_once(DIR_STORAGE.'vendor/autoload.php');` line and it appears to work as designed.

I get having the ability to add non-composer libs or directories. (I notice there is a system and an extension already loaded in the autoloader before vendor.php is called) but for composer stuff, why not just use the composer autoloader?
